src/adrs/0129349218/README.md

```diff
@@ -0,0 +1,532 @@
+The error `message` field should be used to provide a generic description of
+the error condition in English. It is intended for a human audience. Simple
+programs display the message directly to the end-user but the target audience
+is the developer role, strong internationalization is not required, but it is
+recommended.
+
+Developers should never rely on the `message` field to handle the error, but
+instead, they should rely on the `domain` and `type` fields to handle the
+error.
+
+In case services want to provide a more human-readable error message to the
+end-user, they should use the `localized_message` field, which is has the
+message translated to the end-user language, as well as the locale used to
+translate the message.
+
+- Should I always include `localized_message` field?
+
+No, you should only include the `localized_message` field when the error is
+meant to be displayed to the end-user. Most likely the field will be added
+at the edge of the system, aka application layer, where you are aware of the
+end-user actor and the end-user language.
+It is anti-pattern to add the `localized_message` field at the core of the
+system, aka domain layer, where you are not aware of the end-user actor and
+the end-user language, unless you are building a system that is about
+translations.
+
+## Batching and Wrapping Errors
+
+Sometimes errors will happen in a batch, and you will need to wrap them in a
+single error to communicate upstream, or you may need to wrap an error to
+provide more context to the error message.
+
+For example, you want to batch errors that happened related to a command payload
+attributes and wrap them in a single error instead of sending a single error
+per attribute, so you can provide a better user experience.
+
+Or the error happened in a downstream system, and you want to wrap the error
+with more context to the error and have a better understanding of what happened.
+
+The `causes` field should be used to wrap errors. The `causes` field contains a
+list of errors that caused this error.
+
+#### Form/Params/Command/Input Validation
+
+Everything starts with sending a command to the system, therefore, some errors
+are about the validation of such inputs.
+
+Giving feedback to the caller about what exactly went wrong is important to
+achieve clarity and also better user experience when dealing with inputs without
+pushing too much complexity to the clients.
+
+Having a `subject` field that identifies the input that caused the error is
+crucial to achieve such clarity. The `subject` field could be a JSON pointer
+that points to the input that caused the error relative to the command that
+caused the error. That means that when wrapping errors, the `subject` field
+is not absolute, and it is not aware of the wrapping.
+
+For example, imagine the following command:
+
+```js
+const command = {amount: '1000', currency: 'CUP'};
+```
+
+Which will produce the following error:
+
+```js
+const error = {
+  subject: '/data',
```

Comment:
So does /data refer to top-level key? Would be helpful to see an example that uses nesting here.

Reply:
It is a relative JSON Path Pointer based on` whatever Command Message payload structure.

In this case,

```js
const command = {
  data: {
    currency: "INVALID"
  }
}
```